src/vxworks/mod.rs

```diff
@@ -480,6 +474,46 @@ cfg_if! {
                     .finish()
             }
         }
+
+        impl PartialEq for sa_u_t {
```

Comment:
I don't think the traits make sense for the union. You don't know which field is "used" so one can't really do anything meaningful for it. 

What one needs is manual implementations for `pub struct sigaction`, which just treat the union as an `usize` for the purpose of the traits.

Reply:
>  don't think the traits make sense for the union

True, but by implementing them, we avoid "infecting" the structs that use the union. It seems wasteful to have to manually derive these traits for those structs just because they happen to use this type.

Effectively we *are* treating the union as a usize for the purpose of implementing these traits.

Reply:
However if what you suggested is the "standard" approach to using unions, then of course we should follow it for consistency.

Reply:
If these are only used for type punning and accessing both fields is always valid then I think this is probably ok, just please make it explicit by casting to `as usize` in the trait implementations. For the `Debug` impls it does not make much sense to print both variants, I think it is better to just print the contents as an `usize` instead.

Otherwise, I'm not sure if these are really totally-ordered (e.g. if the `Eq` impl is correct).

Reply:
> However if what you suggested is the "standard" approach to using unions, then of course we should follow it for consistency.

The problem with unions is that there is no standard approach: one needs to look at each type and figure out what makes sense for the type. 

Sometimes, the types containing the unions have a field that's used as a discriminant, to denote which field of the union is "active" (Rust does not have such a thing, but C does). So when that happens, one needs to derive the impls manually for those types. For example, for `PartialEq`, only compare the contents of the union fields if the discriminant of both types are the same. For `Debug` only print the active union field according to the discriminant, etc. 

In this case, it appears that the types containing the unions do not have any discriminants, but one should triple check. So if this is right, implementing the traits like an `usize` assumption seems ok.

Reply:
I have updated the code, please let me know if my understanding is wrong.